### PR TITLE
Adjust auto-disable connection logic for spam

### LINKED ISSUE #12287: Adjust auto-disable connection logic to only warn a user once if they meet both warning criteria
If a connection has >=50 consecutive failures in 6 days, and then on the 7th day the connection fails again, the user will get two warning emails. Let's ensure that the user only gets one warning.

### PR DESCRIPTION
**Pull request by terencecho**
## What
address #12287 

tldr: adds logic so if we send a warning for 50 failures in a row, we don't send another warning if there have been only failures in the past 7 days, and vice-versa.

### CODE REVIEW

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/AutoDisableConnectionActivityImpl.java [hunk 1]
```diff
@@ -73,13 +75,16 @@ public AutoDisableConnectionOutput autoDisableFailingConnection(final AutoDisabl
           }
         }
 
+        final boolean warningPreviouslySentForMaxDays =
+            numFailures > 1 && warningPreviouslySentForMaxDays(successTimestamp, maxDaysOfOnlyFailedJobsBeforeWarning, firstJob, jobs);
```
Comment (pmossman) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/AutoDisableConnectionActivityImpl.java:79]: Why do we check `numFailures > 1` here?

Comment (terencecho) [reply airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/AutoDisableConnectionActivityImpl.java:79]: We only send warnings/check for auto-disabling after a failure.

So if this is the first failure (`numFailures == 1`) then there would have been no opportunity to send a warning previously.

reading through this, makes sense to have that logic in `warningPreviouslySentForMaxDays` and i can just pass in `numFailures`

Comment (pmossman) [reply airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/AutoDisableConnectionActivityImpl.java:79]: Cool, that makes sense and yep, agreed that the logic sounds right in `warningPreviouslySentForMaxDays()`

**Review by pmossman — APPROVED** [review]
This makes sense to me, assuming you'll move that ` > 1` check inside the helper!

**Pull request merged**